<launch>
  <arg name="node_name" default="normalized_block_filter" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show edge image" />

  <arg name="kernel_size" default="5" doc="Defines the size of the kernel to be used" />

  <!-- normalized_block_filter.cpp -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="normalized_block_filter" >
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="kernel_size" value="$(arg kernel_size)" />
  </node>
</launch>
